text/0000-sigstore-integration.md

```diff
@@ -0,0 +1,387 @@
+* Build source attestation: adding special modes for public CI systems such as GitHub actions where trusted builders can sign to attest that artifact is built from a particular source.
+
+* Use the public Sigstore instance to manage root keys for TUF.
+* Extending other cargo commands to make use of the information stored in the transparency log when listing dependencies and other places where it makes sense.
+* Integrating with [in-toto.io](https://in-toto.io) to provide more advanced attestation of artifacts..
+
+# Threat model
+
+The threat model is based on the [NPM RFC](https://github.com/npm/rfcs/blob/main/accepted/0049-link-packages-to-source-and-build.md).
+
+Our threat model focuses on a package hijacking attack where an attacker gains access to a crate maintainer's crates.io credentials and uploads a modified version of the crate under a version number that is likely to get pulled in by cargo. Developers that use this crate cannot know whether the crate is built from the original source repository or on the attacker's laptop.
+
+This proposal does not mitigate against compromised crates.io accounts. The aim is to make it harder to execute these types of attacks by creating a public audit trail for where, how and who published a package. Over time the presence of this information can be enforced.
+
+As such, what we care about in this threat model is whether this provenance information can be forged and whether verification can be bypassed.
+
+The following table highlights the impact of different compromises:
+
+
+| Compromised infrastructure | Forge build provenance | Bypass verification |
+| -------------------------- | ---------------------- | ------------------- |
+| crates.io                  | No                     | Yes                 |
+| User account               | No                     | No                  |
+| CI/CD                      | Yes                    | No                  |
+| Sigstore                   | Yes                    | No                  |
+| Network                    | No                     | Yes                 |
+
+An attacker could forge what goes in the provenance information if the CI/CD provider or Sigstore was compromised. Mitigating these attacks is out of scope in this proposal.
+
+An attacker could bypass verification if the crates.io registry, the user or the network was compromised by not providing provenance information or removing it in transit.
+
+The long-term solution to bypassing verification is enforcing or requiring that all crates have provenance information set during install time using cargo. It might take years to get to the point where a large portion of crates have this information set so we'll need to find ways to get there gradually.
+
+Maintainers and developers consuming crates could also opt-in to requiring all their dependencies to include provenance information once their dependencies include this.
+
+# Glossary
```

Comment:
Please define terms when they are used rather than relying on a glossary, or put the glossary at the beginning.